frontend/src/components/Login.jsx

```diff
@@ -0,0 +1,76 @@
+import React from "react";
+import { connect } from "react-redux";
+
+import proxy from "../proxy.js";
+
+class UnconnectedLogin extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { username: "", password: "", loginError: "" };
+  }
+  submitLogIn = event => {
+    event.preventDefault();
+    let data = new FormData();
+    data.append("username", this.state.username);
+    data.append("password", this.state.password);
+    console.log(this.state.username + this.state.password);
+    fetch(proxy + "/login", {
+      method: "POST",
+      credentials: "include",
+      body: data
+    })
+      .then(res => {
+        return res.text();
+      })
+      .then(resBody => {
+        const parsedBody = JSON.parse(resBody);
+        if (!parsedBody.success) {
+          this.setState({ loginError: parsedBody.result });
+          return;
+        } else {
+          this.setState({ username: "", password: "" });
+          this.props.dispatch({ type: "login" });
+        }
+      });
+  };
+  OnChangeUsername = event => {
+    this.setState({ username: event.target.value });
+  };
+  OnChangePassword = event => {
+    this.setState({ password: event.target.value });
+  };
+  render = () => {
+    return (
+      <div>
+        <p>{this.state.loginError}</p>
+        <form onSubmit={this.submitLogIn}>
+          <label>
+            Username:{" "}
+            <input
+              type={"text"}
+              value={this.state.username}
+              onChange={this.OnChangeUsername}
+              name={"username"}
+            />
+          </label>
+          <label>
+            Password:
+            <input
+              type={"password"}
+              value={this.state.password}
+              onChange={this.OnChangePassword}
+              name={"password"}
+            />
+          </label>
+          <input type={"submit"} value={"submit"} />
```

Comment:
So would this! Almost all of my projects wind up with a `<Button>` component, and it usually starts as:

```js
import React from 'react';
import './Button.css';

const Button = (props) => (
  <button className="btn" {...props} />
)

export default Button;
```

Sometimes it gets more complicated: you can have a finite set of styles ("primary", "secondary"), sometimes they have icons, sometimes they should also serve as links... but I've definitely had components that were this simple. It's good because it ensures every button has all the same styles across the app, you don't have to fuss with classNames or worry about what happens if the component gets more complex / needs an icon / etc.